sdk/go/auto/workspace.go

```diff
@@ -129,6 +129,9 @@ type Workspace interface {
 	// ListStacks returns all Stacks created under the current Project.
 	// This queries underlying backend and may return stacks not present in the Workspace.
 	ListStacks(context.Context) ([]StackSummary, error)
```

Comment:
```suggestion
	ListStacks(context.Context, ...optlist.Option) ([]StackSummary, error)
```

I'm wondering if rather than adding a whole new method to list stacks it would be better to add an options argument, similar to what we have in other methods, e.g. `RemoveStack`.  Having a variadic function here should be backwards compatible, and I think we have similar backwards compatible ways of doing this in python and nodejs.  Otherwise this will get trickier to update in the future, if we want to add other potential options here, e.g. filtering by organization/project/tag, or any other options we might add to `stack ls` in the future.

Reply:
I got the golang added and tested, but how should I handle nodejs? It doesn't look like historically we have used interfaces for the params but rather just added optional parameters. I like the following better for readability but don't want to break paradigms unless we are accepting of that.
```
interface ListStacksOptions {
    all?: boolean;
} 
```

Reply:
I think we should be following existing patterns wherever possible.  That is the least surprising and easiest to follow for the user, so I would suggest doing optional parameters for nodejs and python.

Reply:
Hmm, perhaps I spoke too soon. Looking closely at https://github.com/pulumi/pulumi/blob/master/sdk/go/auto/workspace.go and comparing with https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/workspace.ts it appears only RemoveStack is the other function to include variadic option args in workspace.go and just wasn't added to nodejs or python. If it had I think there would be equal argument for adding it in since it has already been done before. Any thoughts on that and what would be most idiomatic for Typescript and Python given that info?

Reply:
I think we need to look a little bit further into methods that do take arguments in that case.  I think `up`/`preview` etc. are a good example here that everyone will be using.  Go uses [variadic arguments](https://github.com/pulumi/pulumi/blob/master/sdk/go/auto/stack.go#L355) there, Nodejs an [optional struct](https://github.com/pulumi/pulumi/blob/master/sdk/nodejs/automation/stack.ts#L150), and Python [optional arguments](https://github.com/pulumi/pulumi/blob/master/sdk/python/lib/pulumi/automation/_stack.py#L206). It's probably best to follow the same pattern here for each language.

Reply:
Makes sense to me. I will follow that pattern for this and get the changes pushed up. I will also open an issue for the other functions to at least get that knowledge tracked somewhere.